app/controllers/users_controller.rb

```diff
@@ -0,0 +1,32 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    # @user = User.new(username: params[:username], email: params[:email], password: params[:password])
+    @user = User.new(user_params)
+    if @user.save
+      redirect_to new_user_path
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update(user_params)
+      redirect_to edit_user_path
+    else
+      render :edit
+    end
+  end
+
+  def user_params
+    params.require(:user).permit(:email, :password, :username)
+  end
```

Comment:
- **_This method should be private since it is only used within the user controller class._**
```
private

def user_params
```